keeper/query_server_vaults.go

```diff
@@ -69,6 +71,82 @@ func (k vaultsQueryServer) PositionsByProvider(ctx context.Context, req *vaults.
 	}, nil
 }
 
+func (k vaultsQueryServer) PendingRewards(ctx context.Context, req *vaults.QueryPendingRewards) (*vaults.QueryPendingRewardsResponse, error) {
+	if req == nil {
+		return nil, types.ErrInvalidRequest
+	}
+
+	rewards, err := k.GetVaultsRewards(ctx)
+	if err != nil {
+		return &vaults.QueryPendingRewardsResponse{
+			PendingRewards: math.ZeroInt(),
+		}, nil
+	}
+
+	totalPendingRewards := math.ZeroInt()
+	for _, reward := range rewards {
+		totalPendingRewards = totalPendingRewards.Add(reward.Rewards)
+	}
+
+	return &vaults.QueryPendingRewardsResponse{
+		PendingRewards: totalPendingRewards,
+	}, nil
+}
+
+func (k vaultsQueryServer) PendingRewardsByProvider(ctx context.Context, req *vaults.QueryPendingRewardsByProvider) (*vaults.QueryPendingRewardsByProviderResponse, error) {
+	if req == nil {
+		return nil, types.ErrInvalidRequest
+	}
+
+	addr, err := k.address.StringToBytes(req.Provider)
+	if err != nil {
+		return nil, types.ErrInvalidRequest
```

Comment:
Let's align this with the other error responses in the module!

```suggestion
		return nil, errors.Wrapf(err, "unable to decode account %s", req.Account)
```